src/subprocess-posix.cc

```diff
@@ -90,9 +88,12 @@ bool Subprocess::Start(SubprocessSet* set, const string& command) {
         // Now can use stderr for errors.
         error_pipe = 2;
         close(output_pipe[1]);
+      } else {
+        // In the console case, the pipe is not needed by the child
+        // process at all. Close it in the child, but keep it open
+        // in the parent in case execl fails.
+        SetCloseOnExec(output_pipe[1]);
```

Comment:
I think this will mean that Ninja will be unable to start tasks in parallel while the console pool task is running, as it will be stuck in the `waitpid` call in `Subprocess::Finish`. Maybe we should be watching for SIGCHLD from pool=console child processes instead of waiting on the pipe.